### PR TITLE
Bagajjal/fix minor issues

### PR DESCRIPTION
**Pull request by bagajjal**
1) Remove unnecessary references to ssh-pkcs11.c
2) fix warnings
3) Create the duplicate token used for create process(ssh-pkcs11-helper.exe)
4) Delete and overwrite the ssh-pkcs11 keys when ssh-add.exe is rerun again.

### CODE REVIEW

**Comment by yan4321** [comment]
@bagajjal , I think you might have missed my comment [here](https://github.com/PowerShell/openssh-portable/pull/537#discussion_r806412585).

Failing to add a provider if it already exists is done intentionally as this is the upstream behavior of [openssh/openssh-portable](https://github.com/openssh/openssh-portable). It also makes sense for the user to _explicitly_ remove a provider that already exists before re-adding.

File: ssh-pkcs11-client.c [hunk 1]
```diff
@@ -251,10 +243,20 @@ pkcs11_terminate(void)
 		pkcs11_del_provider(p->name);
 		TAILQ_REMOVE(&pkcs11_providers, p, next);
 	}
-	pkcs11_terminate_helper();
```
Comment (yan4321) [inline ssh-pkcs11-client.c:254]: This is needed. Have you fully tested it after this change?

File: contrib/win32/win32compat/ssh-agent/agent.c [hunk 2]
```diff
@@ -37,7 +38,7 @@
 #define BUFSIZE 5 * 1024
 
 char* sshagent_con_username;
-int sshagent_client_pid;
+HANDLE sshagent_client_primary_token;
```
Comment (yan4321) [inline contrib/win32/win32compat/ssh-agent/agent.c:41]: @bagajjal , Exposing the token itself as a _global variable_ is a bit riskier than exposing the client's pid.

**Comment by bagajjal** [comment]
> @bagajjal , I think you might have missed my comment [here](https://github.com/PowerShell/openssh-portable/pull/537#discussion_r806412585).
> 
> Failing to add a provider if it already exists is done intentionally as this is the upstream behavior of [openssh/openssh-portable](https://github.com/openssh/openssh-portable). It also makes sense for the user to _explicitly_ remove a provider that already exists before re-adding.

It was an intentional change. To me it doesn't make sense to fail it. If we fail, the error message is not clear as to why it's failing. I don't see failure message in the upstream [code](https://github.com/openssh/openssh-portable/blob/bc16667b4a1c3cad7029304853c143a32ae04bd4/ssh-agent.c#L1419).
Even if it's a deviation from the upstream, I would like to do whatever it's better for customers.

File: contrib/win32/win32compat/ssh-agent/agent.c (hunk 2 above)
Comment (bagajjal) [reply contrib/win32/win32compat/ssh-agent/agent.c:41]: > @bagajjal , Exposing the token itself as a _global variable_ is a bit riskier than exposing the client's pid.

We already do this in multiple places. I don't see why it's riskier.

https://github.com/PowerShell/openssh-portable/blob/3a33ea8deed8a11245202eb442e3331a8f2793e3/contrib/win32/win32compat/spawn-ext.c#L11

**Pull request merged**